frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts

```diff
@@ -1,6 +1,25 @@
 import { RouteComponentProps } from 'react-router';
 import { Extension, ExtensionDeclaration, CodeRef } from '../types';
 
+type ResourcePage = {
+  model: {
```

Comment:
We have multiple declarations of type literal

```ts
model: {
  group: string;
  version: string;
  kind: string;
};
```

as well as this one

```ts
model: {
  group: string;
  version?: string;
  kind?: string;
};
```

I think it's best to create a helper type in `packages/console-dynamic-plugin-sdk/src/api/common-types.ts` to represent a k8s model/resource and use that one for better consistency.

This will also keep the generated JSON schema simpler (references to type definitions, instead of inlining those definitions everywhere they're used).